packages/as-types/assembly/argument.ts

```diff
@@ -245,14 +250,39 @@ export class Args {
    * @return {Result<bool>}
    */
   nextBool(): Result<bool> {
-    if (this.offset + sizeof<u8>() >= this.serialized.length) {
+    if (this.offset + sizeof<u8>() > this.serialized.length) {
       return new Result(
         false,
         "can't deserialize bool from given argument: out of range",
       );
     }
 
-    return new Result(this.serialized[this.offset++] == 0x01);
+    return new Result(!!this.serialized[this.offset++]);
+  }
+
+  /**
+   * Returns the deserialized address
+   *
+   * @return {Result<Address>}
+   */
+  nextAddress(): Result<Address> {
```

Comment:
What about somethink like

```typescript
  next<T>(): Result<T> {
    if (T implements Serializable) {
      return T.deserializa();
    } else {
      ERROR('Object given do not implement Serializable interface'); // this will fail the compilation
    }
```